### PR TITLE
Sanlouise 10330 update connected apps

### PR DESCRIPTION
**Pull request by sanlouise**
[TICKET](https://github.com/department-of-veterans-affairs/va.gov-team/issues/10330)

## Description

1. Add 4 helpful links to the Connected Apps section: 
- FAQ link text: What other third-party apps can I connect to my profile?
Here, we only display apps that the user has not yet connected to
- FAQ link text: How do I connect a third-party app to my profile?
- FAQ link text: What should I do if my records are wrong or out of date in a connected app?
- FAQ link text: What should I do if I no longer trust a connected app?

2. Update copy in Disconnect Modal
3. Update copy in deleted app alert
4. Update Empty State to list all available apps

To those reviewing, this design change is intentional: 
image

## Testing done
Works locally

## Screenshots

**With connected apps**
image

**Without connected apps**
image

## Acceptance criteria
- [x] Add 4 helpful links to the Connected Apps section
- [x] Update copy in Disconnect Modal
- [x] Update copy in deleted app alert
- [x] Update Empty State to list all available apps

## Definition of done
- [ ] Events are logged appropriately
- [ ] Documentation has been updated, if applicable
- [x] A link has been provided to the originating GitHub issue (or connected to it via ZenHub)
- [x] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs

### CODE REVIEW

File: src/applications/personalization/profile360/util/connected-apps.js [hunk 1]
```diff
@@ -3,7 +3,7 @@ export const mockConnectedApps = [
     id: '0oa3s6dlvxgsZr62p2p7',
     type: 'okta_redis_apps',
     attributes: {
-      title: 'Test App',
+      title: 'Apple Health',
```
Comment (sanlouise) [inline src/applications/personalization/profile360/util/connected-apps.js:6]: I changed this to a name that's included in the available connected apps to test the filtering function (this one should not appear under 'apps to connect to'

File: src/applications/personalization/profile-2/components/connected-apps/AdditionalInfoSections.jsx [hunk 2]
```diff
@@ -0,0 +1,159 @@
+import React, { useEffect } from 'react';
```
Comment (erikphansen) [inline src/applications/personalization/profile-2/components/connected-apps/AdditionalInfoSections.jsx:1]: import ordering nit: should put 3rd party stuff first (including lodash), then our own `@department-of-veteran-affairs` stuff, then `platform` or `applications` stuff, and finally relative imports

File: src/applications/personalization/profile-2/components/connected-apps/AdditionalInfoSections.jsx [hunk 3]
```diff
@@ -0,0 +1,159 @@
...
+    : [];
+
+  // This filters out any availableConnectedApps that the user has already connected to
+  const filteredApps = availableConnectedApps.filter(
+    app => !connectedAppsNames.includes(app.name),
+  );
+
+  return (
+    <>
+      {!isEmpty(activeApps) &&
```
Comment (erikphansen) [inline src/applications/personalization/profile-2/components/connected-apps/AdditionalInfoSections.jsx:27]: Ideas for refactoring this: @rianfowler introduced me to the idea of putting this kind of logic outside of the actual render function in aptly-named vars. So you might have something like `const showWhatElseCanIAddInfo = !isEmpty(activeApps) && !isEmpty(filteredApps)` declared above this and then this part of the function reads a little nicer. The idea basically being the `return` part of the functional component should be very clean and easy to read. You could take this further by abstracting away the actual JSX for a given case. So in the end this could read like:
```
{showWhatElseCanIAdd && <WhatElseCanIAddInfo>}
```

File: src/applications/personalization/profile-2/components/connected-apps/ConnectedApps.jsx [hunk 4]
```diff
@@ -10,6 +10,9 @@ import {
   dismissDeletedAppAlert,
   loadConnectedApps,
 } from 'applications/personalization/profile-2/components/connected-apps/actions';
+import { AdditionalInfoSections } from './AdditionalInfoSections';
+import availableConnectedApps from './availableConnectedApps';
+import { isEmpty } from 'lodash';
```
Comment (erikphansen) [inline src/applications/personalization/profile-2/components/connected-apps/ConnectedApps.jsx:15]: nit: move this up with 3rd party imports

File: src/applications/personalization/profile-2/components/connected-apps/ConnectedApps.jsx [hunk 5]
```diff
@@ -44,22 +47,47 @@ export class ConnectedApps extends Component {
...
 
         {!apps ||
-          (allAppsDeleted && (
-            <p className="va-introtext vads-u-font-size--md">
-              You don’t currently have any third-party apps or websites
-              connected to your Veteran data, like health or service records.
-              When you do, this is where you can manage them.
-            </p>
-          ))}
+          (allAppsDeleted &&
```
Comment (erikphansen) [inline src/applications/personalization/profile-2/components/connected-apps/ConnectedApps.jsx:57]: This is another example of something that would be nice to clean up. I can be hard to follow the logic here and would be a little easier if all this conditional stuff was done outside of the `render` method.

**Review by erikphansen — COMMENTED** [review]
Added some thoughts...

File: src/applications/personalization/profile-2/components/connected-apps/AdditionalInfoSections.jsx (hunk 3 above)
Comment (sanlouise) [reply src/applications/personalization/profile-2/components/connected-apps/AdditionalInfoSections.jsx:27]: I 100% agree! All about that clear variable naming 👍  💯

Comment (sanlouise) [reply src/applications/personalization/profile-2/components/connected-apps/AdditionalInfoSections.jsx]: I did assign those conditional combinations to new variables, but left them within the render as the values need to be computed within the render

File: src/applications/personalization/profile-2/components/connected-apps/AdditionalInfoSections.jsx [hunk 6]
```diff
@@ -0,0 +1,160 @@
...
+                information, but isn’t using it in its interface. It’s nothing
+                to worry about. If you have questions about this, send feedback
+                directly to the app.
+              </li>
+            </ul>
+          </AdditionalInfo>
+        </div>
+      )}
+
+      {!isEmpty(activeApps) && (
```
Comment (erikphansen) [inline src/applications/personalization/profile-2/components/connected-apps/AdditionalInfoSections.jsx:131]: oh and it looks like this conditional is the same as line 85 so these can be combined?

File: src/applications/personalization/profile-2/components/connected-apps/AppDeletedAlert.jsx [hunk 7]
```diff
@@ -11,15 +11,13 @@ export const AppDeletedAlert = props => {
   }, []);
 
   const { id, title, dismissAlert } = props;
-  const alertMessage = `${title} won’t be able to see any new information
-  about you from VA, but may still have access to information that was previously
-  shared. To remove any stored data, contact ${title} and request permanent deletion`;
-
+  const alertMessage = `Weve disconnected ${title}. This app can’t access any new information from your VA.gov profile. Some apps may still store information you’ve already shared. If you’d like to ask the app to delete any stored information, contact the app’s support.`;
```
Comment (erikphansen) [inline src/applications/personalization/profile-2/components/connected-apps/AppDeletedAlert.jsx:14]: typo: `Weve`

File: src/applications/personalization/profile-2/components/connected-apps/ConnectedAppDeleteModal.jsx [hunk 8]
```diff
@@ -16,12 +16,13 @@ export function ConnectedAppDeleteModal({
...
-      title="Do you want to disconnect this app?"
+      title={`Please confirm that you want to disconnect ${title}`}
       visible={modalOpen}
+      status="warning"
     >
       <p>
-        {title} won’t have access to new information about you from VA once you
-        disconnect. This may impact the usefulness of the app.
+        Once you disconnect this app, it won’t have access to new information
+        from your VA.gov profile. This may affect how useful the app is to you.
```
Comment (erikphansen) [inline src/applications/personalization/profile-2/components/connected-apps/ConnectedAppDeleteModal.jsx:25]: just want to make sure that `profile` shouldn't be `Profile`

File: src/applications/personalization/profile-2/components/connected-apps/ConnectedApps.jsx [hunk 9]
```diff
@@ -30,6 +33,8 @@ export class ConnectedApps extends Component {
     const activeApps = apps ? apps.filter(app => !app.deleted) : [];
 
     const allAppsDeleted = deletedApps?.length === apps?.length;
+    const showHasNoConnectedApps = !apps || (allAppsDeleted && !loading);
+    const showHasConnectedApps = apps && !allAppsDeleted;
```
Comment (erikphansen) [inline src/applications/personalization/profile-2/components/connected-apps/ConnectedApps.jsx:37]: 🎉

**Review by erikphansen — APPROVED** [review]
Looks good! Just a couple small nits/questions.

File: src/applications/personalization/profile-2/components/connected-apps/ConnectedAppDeleteModal.jsx (hunk 8 above)
Comment (sanlouise) [reply src/applications/personalization/profile-2/components/connected-apps/ConnectedAppDeleteModal.jsx:25]: This is correct 👍
